docs/resiliency/index.asciidoc

```diff
@@ -156,24 +159,60 @@ Almost all files in Elasticsearch now have checksums which are validated before
+
+
+[float]
+=== Take filter cache key size into account (STATUS: ONGOING)
+
+Commonly used filters are cached in Elasticsearch. That cache is limited in size (10% of node's memory by default) and is being evicted based on a least recently used policy. The amount of memory used by the cache depends on two primary components - the values it stores and the keys associated with them. Calculating the memory footprint of the values is easy enough but the keys accounting is trickier to achieve as they are, by default, raw Lucene objects. This is largely not a problem as the keys are dominated by the values. However, recent optimizations in Lucene have changed the balance causing the filter cache to grow beyond it's size.
+
+While we are working on a longer term solution ({GIT}9176[#9176]), we introduced a minimum weight of 1k for each cache entry. This puts an effective limit on the number of entries in the cache. See {GIT}8304[#8304] (STATUS: DONE, fixed in v1.4.0)
+
+== Completed
 
 [float]
-=== Prevent setting minimum_master_nodes to more than the current node count (STATUS: ONGOING)
+=== Ensure cluster state ID is incremental (STATUS: DONE, v1.5.1)
+
+It is possible that the current state ID can be reset or even go backwards
+when nodes or the cluster are restarted, or nodes with legacy data join
+the cluster.  Elasticsearch now ensures that the state ID always moves
+forwards, and throws an exception when a legacy ID is higher than the
+current ID.  See {GIT}10316[#10316] (STATUS: DONE, v1.5.1)
+
+[float]
+=== Verification of index UUIDs (STATUS: DONE, v1.5.0)
+
+When deleting and recreating indices rapidly, it is possible that cluster state
+updates can arrive out of sync and old states can be merged incorrectly.  Instead,
+Elasticsearch now checks the index UUID to ensure that cluster state updates
+refer to the same index version that is present on the local node.
+See {GIT}9541[#9541] and {GIT}10200[#10200] (STATUS: DONE, Fixed in v1.5.0)
+
+[float]
+=== Disable recovery from known buggy versions (STATUS: DONE, v1.5.0)
+
+Corruptions have been known to occur when doing a rolling restart from older, buggy versions.
+Now, shards from versions before v1.4.0 are copied over in full and recovery from versions
+before v1.3.2 are disabled entirely. See {GIT}9925[#9925] (STATUS: DONE, Fixed in v1.5.0)
+
+
+[float]
+=== Upgrade 3.x segments on engine startup (STATUS: DONE, v1.5.0)
+
+Upgrading old 3.x segments can be error prone and can result in corruption when merges are being
+run concurrently. Instead, Elasticsearch will now upgrade 3.x segments in place before the
+engine starts.  See {GIT}9899[#9899] (STATUS; DONE, fixed in v1.5.0)
```

Comment:
I think this is misleading, its not upgrading the segments, it just writes a new commit point using the new version